jvm/src/main/kotlin/app/cash/trifle/Certificate.kt

```diff
@@ -48,35 +48,20 @@ data class Certificate internal constructor(
    * @param date - The date to use for verification against certificates' validity windows. If null,
    *   the current time is used.
    *
-   * @return - VerifyResult status indicating issue if certificate is not signed by the given root
-   *   certificate, or if the information present doesn't match that of the certificateRequest.
-   *   SUCCESS otherwise.
+   * @return - Boolean status indicating error in validation or success.
+   *   Exception thrown if certificate is not signed by the given root certificate, or
+   *   certificate is expired or invalid or
+   *   if the information present doesn't match that of the certificateRequest.
    */
   fun verify(
     certificateRequest: CertificateRequest,
     ancestorCertificateChain: List<Certificate>,
     anchorCertificate: Certificate,
     date: Date? = null
-  ): VerifyResult {
+  ): Boolean {
```

Comment:
If you're planning to replace `VerifyResult`, consider cleaning up the data class (i.e remove it).

Reply:
Good idea about removing it.